event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/core/EventsService.java

```diff
@@ -1,90 +1,90 @@
-import uk.gov.justice.services.eventsourcing.repository.jdbc.event.Event;
-import uk.gov.justice.services.eventsourcing.repository.jdbc.event.EventJdbcRepository;
+import uk.gov.justice.services.eventsourcing.source.core.EventSource;
+import uk.gov.justice.services.eventsourcing.source.core.EventStream;
+import uk.gov.justice.services.messaging.JsonEnvelope;
 
 import java.time.ZonedDateTime;
 import java.util.List;
 import java.util.UUID;
 import java.util.function.Function;
+import java.util.stream.Stream;
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
-import javax.json.JsonObject;
 
 @ApplicationScoped
 public class EventsService {
 
     @Inject
-    private EventJdbcRepository repository;
+    private EventSource eventSource;
 
     public List<EventEntry> events(final UUID streamId,
                                    final Position position,
                                    final Direction direction,
                                    final long pageSize) {
 
-        final List<Event> eventEntries = entitiesPage(streamId, position, direction, pageSize);
-        return eventEntries(eventEntries);
-    }
+        final EventStream eventStream = eventSource.getStreamById(streamId);
 
-    private List<Event> entitiesPage(
-            final UUID streamId,
-            final Position position,
-            final Direction direction,
-            final long pageSize) {
         if (position.isHead()) {
-            return repository.head(streamId, pageSize).collect(toList());
+            final long versionHead = eventStream.getCurrentVersion() - pageSize + 1;
+            final Stream<JsonEnvelope> events = eventStream.readFrom(versionHead).limit(pageSize);
+            return reverse(eventEntries(events));
         }
 
         if (position.isFirst()) {
-            return repository.first(streamId, pageSize).collect(toList());
+            final int versionFirst = 1;
+            final Stream<JsonEnvelope> events = eventStream.readFrom(versionFirst).limit(pageSize);
+            return reverse(eventEntries(events));
         }
 
         if (FORWARD.equals(direction)) {
-            return repository.forward(streamId, position.getSequenceId(), pageSize).collect(toList());
+            final long version = position.getSequenceId();
+            final Stream<JsonEnvelope> events = eventStream.readFrom(version).limit(pageSize);
+
+            return reverse(eventEntries(events));
         }
 
         if (BACKWARD.equals(direction)) {
-            return repository.backward(streamId, position.getSequenceId(), pageSize).collect(toList());
+            final long version = position.getSequenceId() - pageSize + 1;
+            final Stream<JsonEnvelope> events = eventStream.readFrom(version).limit(pageSize);
+            return reverse(eventEntries(events));
         }
-
         return emptyList();
     }
 
     public boolean recordExists(final UUID streamId, final long sequenceId) {
```

Comment:
again, recordExists is the correct name?

Should we be doing this all in java or is it just a query?

Reply:
changed to eventExists